mlpp_lib/standardizers.py

```diff
@@ -11,49 +12,269 @@
+            self.all_vars.extend(remaining_var)
+
+        for i in range(len(self.parameters)):
+            normalizer, variables, inputs = self.parameters[i]
+            normalizer.fit(dataset=dataset, variables = variables, dims = dims)
+            self.parameters[i] = (normalizer, variables, inputs)
+
+    
+    def transform(self, *datasets: xr.Dataset) -> tuple[xr.Dataset, ...]:
+        
+        for parameter in self.parameters:
+            method, variables, _ = parameter
+            variables = list(set(variables) & set(datasets[0].data_vars)) # ensure that only variables that are in the dataset are transformed
+            datasets = method.transform(*datasets, variables=variables)
+        return datasets
+    
+    
+    def inverse_transform(self, *datasets: xr.Dataset) -> xr.Dataset:
+        
+        for parameter in self.parameters:
+            method, variables, _ = parameter
+            variables = list(set(variables) & set(datasets[0].data_vars)) # ensure that only variables that are in the dataset are inv-transformed
+            datasets = method.inverse_transform(*datasets, variables=variables)
+
+        return datasets
+    
+    
+    @classmethod
+    def from_dict(cls, in_dict: dict) -> Self:
+        method_var_dict = {}
+
+        # check whether dict corresponds to old Standardizer format
+        first_key = list(in_dict.keys())[0]
+        if first_key not in [cls.__name__ for cls in Normalization.__subclasses__()]:
+            subclass = Standardizer().from_dict(in_dict)
+            inputs = {"mean": subclass.mean, "std": subclass.std, "fillvalue": subclass.fillvalue}
+            method_var_dict[subclass.name] = (list(subclass.mean.data_vars), inputs)
+        else:
+            for method_name, inner_dict in in_dict.items():
+                tmp_class = create_normalization_from_str(method_name).from_dict(inner_dict)
+                inputs = {key: getattr(tmp_class, key) for key in inner_dict if getattr(tmp_class, key, None) is not None}
+                method_var_dict[tmp_class.name] = (inner_dict["channels"], inputs)
+        return cls(method_var_dict)
+    
+    
+    def to_dict(self):
+        out_dict = {}
+        for parameter in self.parameters:
+            method, variables, _ = parameter
+            out_dict_tmp = method.to_dict()
+            out_dict_tmp["channels"] = variables
+            out_dict[method.name] = out_dict_tmp
+        return out_dict
+    
+    @classmethod
+    def from_json(cls, in_fn: str) -> Self:
+        with open(in_fn, "r") as f:
+            in_dict = json.load(f)
+        return cls.from_dict(in_dict)
+    
+    def save_json(self, out_fn: str) -> None:
+        out_dict = self.to_dict()
+        if len(out_dict) == 0 or out_dict[list(out_dict.keys())[0]] is None:
+            raise ValueError(f"{self.name} wasn't fit to data")
+        with open(out_fn, "w") as outfile:
+            json.dump(out_dict, outfile, indent=4)
+
+class Normalization:
```

Comment:
new idea (by looking at the names below): what about naming this "Scaler"?

Reply:
At the moment, it makes some sense to name it Scaler, but I fear in the future, it may not be as coherent as right now. However, renaming it "DataTranformer" for instance would ensure the compatibility even with future methods, if we want to rename it (and rename "Normalization" to Transformation/transform/DataTransform ...)